sdks/java/core/src/main/java/org/apache/beam/sdk/io/PubsubIO.java

```diff
@@ -630,9 +658,9 @@ private Bound(String name, PubsubSubscription subscription, PubsubTopic topic,
                       .setCoder(coder);
         } else {
           @Nullable ProjectPath projectPath =
-              topic == null ? null : PubsubClient.projectPathFromId(topic.project);
-          @Nullable TopicPath topicPath =
-              topic == null ? null : PubsubClient.topicPathFromName(topic.project, topic.topic);
+              topic == null ? null : PubsubClient.projectPathFromId(topic.get().project);
```

Comment:
Yep.  Was checking that we liked the general NVP pattern first, now fixing this.